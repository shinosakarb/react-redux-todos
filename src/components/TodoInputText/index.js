import React, { Component } from 'react'

class TodoInputText extends Component {
  constructor(props) {
    super(props)
    this.state = {
      text: ''
    }
  }

  handleSubmit = e => {
    const text = e.target.value.trim()
    if (e.which === 13) {
      this.props.onSave(text)
      this.setState({ text: '' })
    }
  }

  handleChange = e => {
    this.setState({ text: e.target.value })
  }

  render() {
    return (
      <input
        type="text"
        value={this.state.text}
        onChange={this.handleChange}
        onKeyDown={this.handleSubmit}
      />
    )
  }
}

export default TodoInputText
