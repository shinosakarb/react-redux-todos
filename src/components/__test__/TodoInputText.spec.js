import React from 'react'
import { shallow, mount } from 'enzyme'
import toJson from 'enzyme-to-json'
import TodoInputText from '../TodoInputText'

describe('TodoInputText', () => {
  it('handleChange', () => {
    const onSave = jest.fn()
    const wrapper = mount(<TodoInputText onSave={onSave} />)

    const inputText = wrapper.find('input')

    // change input text
    inputText.simulate('change', { target: { value: 'todo1' } })
    inputText.simulate('keyDown', { which: 13 })

    expect(toJson(wrapper)).toMatchSnapshot()
    expect(onSave.mock.calls).toHaveLength(1)
  })
})
