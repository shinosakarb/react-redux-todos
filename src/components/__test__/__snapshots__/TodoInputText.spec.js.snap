// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoInputText handleChange 1`] = `
<TodoInputText
  onSave={[Function]}
>
  <input
    onChange={[Function]}
    onKeyDown={[Function]}
    type="text"
    value=""
  />
</TodoInputText>
`;
